js-peer/src/components/peer.tsx

```diff
@@ -0,0 +1,190 @@
+import Blockies from 'react-18-blockies'
+import { classNames } from "@/lib/classes"
+
+interface DMMenuItemProps {
+  peerId: PeerId
+}
+
+function DMMenuItem({ peerId }: DMMenuItemProps) {
+  const { setRoomId } = useChatContext()
+
+  const handleSetRoomId = () => {
+    setRoomId(peerId.toString())
+  }
+
+  return (
+    <MenuItem>
+      {({ focus }) => (
+        <span
+          className={classNames(
+            focus ? 'bg-gray-100 text-gray-900' : 'text-gray-700',
+            'block px-4 py-2 text-sm',
+          )}
+          onClick={() => handleSetRoomId()}
+        >
+          Direct Message
+        </span>
+      )}
+    </MenuItem>
+  )
+}
+
+interface UnsupportedMenuItemProps {
+  text: string
+}
+
+function UnsupportedMenuItem({ text }: UnsupportedMenuItemProps) {
+  return (
+    <MenuItem>
+      {({ focus }) => (
+        <span
+          className={classNames(
+            focus ? 'bg-gray-100 text-gray-900' : 'text-gray-700',
+            'block px-4 py-2 text-sm',
+          )}
+        >
+          {text}
+        </span>
+      )}
+    </MenuItem>
+  )
+}
+
+interface UnidentifiedMenuItemProps {
+  peerId: PeerId
+  state: {
+    dialing: boolean
+    setDialing: (dialing: boolean) => void
+    error: string
+    setError: (error: string) => void
+  }
+}
+
+function UnidentifiedMenuItem({ peerId, state }: UnidentifiedMenuItemProps) {
+  const { libp2p } = useLibp2pContext()
+
+  const handleQuery = async () => {
+    try {
+      state.setError('')
+      state.setDialing(true)
+
+      await libp2p.dial(peerId, { signal: AbortSignal.timeout(5000) })
+    } catch (e) {
+      state.setError(`${e}`)
+      console.error('Failed to dial', e)
+    } finally {
+      state.setDialing(false)
+    }
+  }
+
+  return (
+    <MenuItem>
+      {({ focus }) => (
+        <span
+          className={classNames(
+            focus ? 'bg-gray-100 text-gray-900' : 'text-gray-700',
+            'block px-4 py-2 text-sm',
+          )}
+          onClick={() => handleQuery()}
+        >
+          {!state.dialing && !state.error && 'Dial'}
+          {state.dialing && 'Dialing'}
+          {!state.dialing && state.error && `Retry (${state.error})`}
+        </span>
+      )}
+    </MenuItem>
+  )
+}
+
+export interface PeerProps {
+  peer: PeerId,
+  self: boolean,
+  withName: boolean,
+  withUnread: boolean
+}
+
+export function Peer({ peer, self, withName, withUnread }: PeerProps) {
+  const { libp2p } = useLibp2pContext()
+  const { directMessages } = useChatContext()
+  const [dialing, setDialing] = useState(false)
+  const [error, setError] = useState('')
+  const [identified, setIdentified] = useState(false)
+
+  useEffect(() => {
+    const init = async () => {
+      if (await libp2p.peerStore.has(peer)) {
+        const p = await libp2p.peerStore.get(peer)
+        if (p.protocols.length > 0) {
+          setIdentified(true)
+        }
+      }
+    }
+
+    init()
+  }, [libp2p.peerStore, peer])
+
+  return (
```

Comment:
Given that you can know here whether the peer supports DMs (`libp2p.services.directMessage.isDMPeer(peer)` and whether the peer is identified, I would suggest removing the menu item and just making the click on the peer go to a DM. 

It saves the user a click and make the UX a bit slicker. 

What do you think?

Reply:
done 👍